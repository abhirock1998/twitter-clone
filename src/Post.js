import { Avatar } from "@material-ui/core";
import React, { forwardRef } from "react";
import RepeatIcon from "@material-ui/icons/Repeat";
import FavoriteBorderIcon from "@material-ui/icons/FavoriteBorder";
import PublishIcon from "@material-ui/icons/Publish";
import ChatBubbleOutlineIcon from "@material-ui/icons/ChatBubbleOutline";
import VerifiedUserIcon from '@material-ui/icons/VerifiedUser';
import "./post.css";
const Post = forwardRef(
  ({ avatar, name, username, verified, text, image, timestamp }, ref) => {
    return (
      <div data-testid='post' ref={ref} className="post">
        <div className="post_avatar">
          <Avatar src={avatar} />
        </div>
        <div className="post_body">
          <div className="post_header">
            <div className="post_headerText">
              <h3>
                {name}
                <span className="post_userName">
                  {verified && <VerifiedUserIcon className="post_badge" />}@
                  {username?.toLowerCase()}{" "}
                  <small>
                    {timestamp && new Date(timestamp?.toDate()).toUTCString()}
                  </small>
                </span>
              </h3>
            </div>
            <div className="post_headerDescription">
            <div>  <p>{text}</p></div>
            </div>
          </div>
          <img src={image} alt="" />
          <div className="post_footer">
            <ChatBubbleOutlineIcon fontSize="small" />
            <RepeatIcon fontSize="small" />
            <FavoriteBorderIcon fontSize="small" />
            <PublishIcon fontSize="small" />
          </div>
        </div>
      </div>
    );
  }
);

export default Post;
