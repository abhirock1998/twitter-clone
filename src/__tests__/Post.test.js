import React from "react";
import { render, fireEvent, getByTestId } from "@testing-library/react";
import Post from '../Post';


describe("<Post/> ", () => {
  it("render the <Post/> with dummy data", () => {
    const {container,getByTestId,queryByTestId} = render(
       <Post/>
    )
    

    expect(queryByTestId("post")).toBeTruthy();
    fireEvent.click(getByTestId('post'));
    expect(queryByTestId('post')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
   
  });
});
