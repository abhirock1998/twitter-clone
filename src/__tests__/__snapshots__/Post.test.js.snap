// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post/>  render the <Post/> with dummy data 1`] = `
<div
  class="post"
  data-testid="post"
>
  <div
    class="post_avatar"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiAvatar-fallback"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
  </div>
  <div
    class="post_body"
  >
    <div
      class="post_header"
    >
      <div
        class="post_headerText"
      >
        <h3>
          <span
            class="post_userName"
          >
            @
             
            <small />
          </span>
        </h3>
      </div>
      <div
        class="post_headerDescription"
      >
        <div>
            
          <p />
        </div>
      </div>
    </div>
    <img
      alt=""
    />
    <div
      class="post_footer"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H6l-2 2V4h16v12z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 7h10v3l4-4-4-4v3H5v6h2V7zm10 10H7v-3l-4 4 4 4v-3h12v-6h-2v4z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M5 4v2h14V4H5zm0 10h4v6h6v-6h4l-7-7-7 7z"
        />
      </svg>
    </div>
  </div>
</div>
`;
